packages/skygear-core/src/storage.ts

```diff
@@ -1,13 +1,114 @@
-import { StorageDriver, JSONValue } from "./types";
+import { StorageDriver, JSONValue, User, Identity } from "./types";
+
+import {
+  encodeUser,
+  encodeIdentity,
+  decodeUser,
+  decodeIdentity,
+} from "./encoding";
 
 function scopedKey(key: string): string {
   return `skygear2_${key}`;
 }
 
+function keyAccessToken(name: string): string {
+  return `${name}_accessToken`;
+}
+
+function keyUser(name: string): string {
+  return `${name}_user`;
+}
+
+function keyIdentity(name: string): string {
+  return `${name}_identity`;
+}
+
+function keyOAuthRedirectAction(name: string): string {
+  return `${name}_oauthRedirectAction`;
+}
+
 /**
  * @public
  */
 export class ContainerStorage {
```

Comment:
We better introduce an interface `IContainerStorage` (not sure about this name, we do not have convention to name interface with prefix `I`) and let `Container` depend on this interface instead of this class. This allows us to plug a compatible storage in skycli without having to extend `ContainerStorage`.